core/unix/os.c

```diff
@@ -2751,10 +2751,10 @@ static void
 replace_thread_id(thread_id_t old, thread_id_t new)
 {
 #ifdef HAVE_TLS
-    thread_id_t new_tid = new;
+    int32_t new_tid = (int32_t) new; /* can't use thread_id_t since it's 64-bits on x64 */
     ASSERT(is_thread_tls_initialized());
     DOCHECK(1, {
-        thread_id_t old_tid;
+        int32_t old_tid; /* can't use thread_id_t since it's 64-bits on x64 */
```

Comment:
Wait -- os_local_state_t.tid is thread_id_t though, so we need to read a pointer-sized value via READ_TLS_SLOT_IMM, rather than changing these locals to ints.  Maybe have a READ_TLS_TIDSZ_SLOT_IMM or sthg.